### DISCUSSION TITLE
Drag and Drop Tools (and their extensions)

### DISCUSSION THREAD

**Discussion by BradKML** [SmartManoj/Kevin#139] [General]
Here are some of the drag-and-drop LLM tooling that might be good as a baseline on what possible tools would be needed
- https://github.com/langgenius/dify
- https://github.com/langflow-ai/langflow
- https://github.com/infiniflow/ragflow
- https://github.com/FlowiseAI/Flowise

Some questions:
1. Are there some integrations that are inherently more valuable than others?
2. What are the major constraints of pure agentic UX vs DAG-like UX?
3. How can be agents more efficient than DAGs when it comes to semi-established work patterns?

**Comment by SmartManoj** [comment]
How do you compare those with [computer.tldraw.com](https://computer.tldraw.com)?

**Reply by BradKML** [reply]
I am partly thinking about how TLDraw is missing a lot of components from these other drag-and-drops:
- Toolkits https://docs.dify.ai/guides/tools https://docs.langflow.org/components-tools https://docs.flowiseai.com/integrations/langchain/tools
- Vector DB + RAGs + Rerranking https://docs.dify.ai/learn-more/extended-reading/retrieval-augment https://docs.langflow.org/components-vector-stores 
- Agents (not gonna elaborate too much here)

But at the same time TLDraw is only good because it has the kind of hand-drawn chic (aesthetics not functionally) attached to it. But a lot of these other professional GUI tools can also have the vector DB in the command line. And the way RAGs are used can vary a lot, like document management vs long term memory for web browsing. That is a different issue altogether.
